src/routes/courses/[slug]/+page.svelte

```diff
@@ -1,40 +1,27 @@
 <script>
+
   import { onMount } from "svelte";
-  // @ts-ignore
-  import { gql } from "@apollo/client/core/core.cjs";
-  import { query } from "svelte-apollo";
-  import { page } from "$app/stores";
-  import { FETCH_COURSE_BY_SLUG, FETCH_COURSE_BY_COMPOSITE } from "$lib/queries/courses";
   import StartCourseButton from "../../../components/StartCourseButton.svelte";
   
   export let data;
-  const { slug } = data;
-
-  const courseBySlug = query(FETCH_COURSE_BY_SLUG, {
-    variables: { slug },
-  });
+  let  {course, active}  = data;
 
-  const activeCourse = query(FETCH_COURSE_BY_COMPOSITE, { variables: { user_id: "a4f9150e-4619-4b18-807d-2f273b75c12f", course_id: "6477f941-f072-4f1e-95da-b1d02a81ae6f" } });
 
   onMount(async () => {
-    courseBySlug.refetch();
+    
   });
 
-  $: courseBySlug.refetch();
+  $: course, active;
 
 </script>
 
 
-{#if $courseBySlug.data}
-  <div>
-    {#each $courseBySlug.data.courses as course}
-      <div> 
-        <h1>{course.name}</h1>
-        <p>{course.description}</p>
-        <StartCourseButton />
-      </div>
-    {/each}
-  </div>
-{:else}
-  <p>Loading...</p>
-{/if}
+<div>
+    <div> 
+      <h1>{course.name}</h1>
+      <p>{course.description}</p>
+      <StartCourseButton status={active?.status === "Subscribed" ? true : false} course_id={course.id} slug={course.slug} />
```

Comment:
Different usage of status here

Reply:
We expect 3 different values for status here, I think using a map/ object would be the most suitable approach.

Reply:
wyd? status is a string not a boolean

Reply:
> We expect 3 different values for status here, I think using a map/ object would be the most suitable approach.

In this case, is it better to pass the variable itself and do the cases inside the button?

Reply:
Well, we could avoid using a boolean and simply pass the status as is. On the other side, we can have an object with each status value as a key and we could use that to render custom styles and implement custom logic.